src/mito2/src/gc.rs

```diff
@@ -247,7 +282,12 @@ impl LocalGcWorker {
                 .cloned()
                 .unwrap_or_else(HashSet::new);
             let files = self.do_region_gc(region.clone(), &tmp_ref_files).await?;
-            deleted_files.insert(*region_id, files);
+            let index_files = files
+                .iter()
+                .filter_map(|f| f.index_version().map(|v| (f.file_id(), v)))
```

Comment:
if index_version is None, we will skip the index file,  is it expected?

Reply:
> if index_version is None, we will skip the index file, is it expected?

yes, to distinguish between having a index file or not